satpy/writers/core/utils.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2025 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilities shared between writers."""
+
+from xarray.coding.times import decode_cf_datetime
+
+
+def get_valid_time(dataset):
+    """Get the valid time for a dataset.
+
+    If a dataset has time coordinates, get the arithmetic mean time as a
+    representative time for the dataset.
```

Comment:
I wouldn't name this function as "valid" time but "average" or "mean" (observation) time. Or maybe add a kwarg that controls the behaviour? Others might have different definition of "valid" time. The other options could just be `NotImplementedError` for now.

Reply:
Renamed `get_mean_time` in c3626c430.